crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -450,7 +467,13 @@ fn highlight_name(db: &RootDatabase, def: Definition) -> Highlight {
         Definition::Field(_) => HighlightTag::Field,
         Definition::ModuleDef(def) => match def {
             hir::ModuleDef::Module(_) => HighlightTag::Module,
-            hir::ModuleDef::Function(_) => HighlightTag::Function,
+            hir::ModuleDef::Function(func) => {
+                let mut h = HighlightTag::Function.into();
+                if func.is_unsafe(db) {
+                    h |= HighlightModifier::Unsafe;
+                }
+                return h;
+            }
```

Comment:
Wait, so, for actual unsafe highlighting, we don't even need to track unsafe context, we only need types? 

So, we can implement the highlighting bits without changes to lowering and such? Let's split this off into a separate PR then!

Reply:
that makes sense. I'll split that off after work tonight. Should I also implement highlighting within vs code? Something like https://github.com/rust-analyzer/rust-analyzer/pull/4603/files#diff-9234de0bfa99228a67ea76a4391c86f1R598

Reply:
If you add news tags/modifiers, they should be specified in package.json. But it seems like we already have operator and unsafe, so that should not be necessary

Reply:
Opened #4720